src/runtime_src/core/common/query_requests.h

```diff
@@ -1184,6 +1185,50 @@ struct clock_freq_topology_raw : request
   get(const device*) const = 0;
 };
 
+struct xrt_resource_raw : request
+{
+  enum resource_type {
+    ipu_clk_max,
+    ipu_tops_max,
+    ipu_task_max,
+    ipu_tops_curr,
+    ipu_task_curr
+  };
+
+  struct xrt_resource_query {
+      resource_type type;
+      union
+      {
+          uint64_t data_uint64;
+          double data_double;
+      };
+  };
+
+  using result_type = std::vector<xrt_resource_query>; // get value type
+  static const key_type key = key_type::xrt_resource_raw;
+
+  static std::string
+  get_name(xrt_core::query::xrt_resource_raw::resource_type type) {
+      switch (type) {
+      case resource_type::ipu_clk_max:
+          return "Max H-Clocks";
+      case resource_type::ipu_tops_max:
+          return "Max TOPs";
+      case resource_type::ipu_task_max:
+          return "Max Tasks";
+      case resource_type::ipu_tops_curr:
+          return "Current TOPs";
```

Comment:
Do we have this value? I vaguely remember we discussed that we don't have this readily available and that we should start with showing just the Max TOPS. Did something change after that?

Reply:
Yes, the decision still holds. We only return MAX TOPs in add_tops_info@info_platform.cpp. But these resources are usually queried in bunch. the code is written in such that we don't have to come back to refactoring the query resources code to support more resources when they are ready. later we just need to add them to the returned data buffer, that will be a limited change.